### DISCUSSION TITLE
Open /etc/docker/certs.d/ghcr.io: permission denied

### DISCUSSION THREAD

**Discussion by alexellis** [community/community#26880] [Actions]
<p>When building a multi-arch image, the CI job for inlets is pushing each individual image, but failing on the manifest push.</p>
<p>This does work on the Docker Hub, but seems to fail for GHCR:</p>
<pre><code class="lang-auto">docker manifest create --amend ghcr.io/inlets/inlets:2.7.5-2-g9d88c2b ghcr.io/inlets/inlets:2.7.5-2-g9d88c2b-amd64 ghcr.io/inlets/inlets:2.7.5-2-g9d88c2b-arm64 ghcr.io/inlets/inlets:2.7.5-2-g9d88c2b-armhf
open /etc/docker/certs.d/ghcr.io: permission denied
Makefile:61: recipe for target 'manifest-ghcr' failed
</code></pre>
<p><a href="https://travis-ci.com/github/inlets/inlets/builds/182828844#L939" class="onebox" target="_blank" rel="nofollow noopener">https://travis-ci.com/github/inlets/inlets/builds/182828844#L939</a></p>
<p>Sources:</p>
<p><a href="https://github.com/inlets/inlets" rel="nofollow noopener">https://github.com/inlets/inlets</a> - see Makefile and .travis.yml (I can’t include the links, the forum won’t let me)</p>
<p>There was an old issue back in Moby with a similar error with Docker 18.x, so I’ve upgraded to the latest 19.x in Travis, and it still had no effect.</p>
<p>Are manifests supported with GHCR?</p>

**Comment by alexellis** [comment]
<p>Here are two screenshots to show the differences between the Docker Hub and GHCR with regards to the manifests</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e4120aaeb9cedb94ee0ca527a21cdef8ba54a4c1.png" data-download-href="/uploads/short-url/wxBw5dNlXVUDcZhPM4fKawN1lkd.png?dl=1" title="Screenshot 2020-09-04 at 20.41.03" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6358735/181100432-9b3272fd-7031-43f2-9c7e-144d30c5f36c.png" alt="Screenshot 2020-09-04 at 20.41.03" data-base62-sha1="wxBw5dNlXVUDcZhPM4fKawN1lkd" width="690" height="111" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e4120aaeb9cedb94ee0ca527a21cdef8ba54a4c1_2_690x111.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e4120aaeb9cedb94ee0ca527a21cdef8ba54a4c1_2_1035x166.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e4120aaeb9cedb94ee0ca527a21cdef8ba54a4c1_2_1380x222.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e4120aaeb9cedb94ee0ca527a21cdef8ba54a4c1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-09-04 at 20.41.03</span><span class="informations">1972×318 34.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by alexellis** [comment]
<p>The erroring example (again, I can’t upload two screenshots)</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/ef2caa4ff16112cb2feb0c8be73508bef844922a.png" data-download-href="/uploads/short-url/y7POf2hziMwt8g2i86l2nqaPQIG.png?dl=1" title="Screenshot 2020-09-04 at 20.40.54" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6358735/181100438-1794d1c9-1c00-4f5c-91bf-a9c224eee7da.png" alt="Screenshot 2020-09-04 at 20.40.54" data-base62-sha1="y7POf2hziMwt8g2i86l2nqaPQIG" width="690" height="86" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/ef2caa4ff16112cb2feb0c8be73508bef844922a_2_690x86.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/ef2caa4ff16112cb2feb0c8be73508bef844922a_2_1035x129.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/ef2caa4ff16112cb2feb0c8be73508bef844922a_2_1380x172.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/ef2caa4ff16112cb2feb0c8be73508bef844922a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-09-04 at 20.40.54</span><span class="informations">1986×250 25.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by clarkbw** [comment]
<p>I’m using <code>buildx</code> in Actions and I’m not seeing this error. I’ll give it a try to use the same <code>manifest create</code> as you have here.</p>

**Comment by clarkbw** [comment]
<p>This worked for me on my machine… I’m not sure what’s going on with the error you’re seeing unless this is one of those things where the docker daemon needs to be restarted.</p>
<pre><code class="lang-shell">export Version=2.7.5
for Arch in arm64 amd64 armhf; do 
  docker pull inlets/inlets:${Version}-${Arch} 
  docker tag inlets/inlets:${Version}-${Arch} ghcr.io/$USER/inlets:${Version}-${Arch}
  docker push ghcr.io/$USER/inlets:${Version}-${Arch}
done

docker pull inlets/inlets:${Version}
docker tag inlets/inlets:${Version} ghcr.io/$USER/inlets:${Version}
docker push ghcr.io/$USER/inlets:${Version}


docker manifest create --amend ghcr.io/$USER/inlets:${Version} ghcr.io/$USER/inlets:${Version}-amd64 ghcr.io/$USER/inlets:${Version}-arm64 ghcr.io/$USER/inlets:${Version}-armhf

Created manifest list ghcr.io/clarkbw/inlets:2.7.5
</code></pre>
<p>Which created this:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/users/clarkbw/packages/container/inlets/41400" target="_blank" rel="noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2134/181100441-d6a2b830-5d1d-4200-9c71-0f3e439ee9ef.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/users/clarkbw/packages/container/inlets/41400" target="_blank" rel="noopener">Build software better, together</a></h3>

<p>GitHub is where people build software. More than 50 million people use GitHub to discover, fork, and contribute to over 100 million projects.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by clarkbw** [comment]
<pre><code class="lang-auto">docker --version
Docker version 19.03.12, build 48a66213fe
</code></pre>

**Comment by alexellis** [comment, marked as answer]
<p>It seems like we both have the same version of Docker, however the Travis build fails on every build, without fail.</p>
<pre><code class="lang-bash"> Version:           19.03.12
</code></pre>
<p>I don’t know if I could push from my local machine (I probably can, given your output)</p>
<p>I wasn’t about to find much information on Google or the issue tracker on the error:</p>
<pre><code class="lang-bash">ghcr.io/inlets/inlets:2.7.5-2-g9d88c2b-armhf
open /etc/docker/certs.d/ghcr.io: permission denied
Makefile:61: recipe for target 'manifest-ghcr' failed
make: *** [manifest-ghcr] Error 1
</code></pre>
<p>The message about <code>certs.d</code> smells like Docker may not think that <a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a> is a valid TLS certificate, and that it needs to dig up some self-signed cert.</p>
<p>I’m trying to isolate the issue here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/6358735/181100442-43ba80d6-6d4f-4499-a99c-b5b7f9c3966e.png" class="site-icon" width="32" height="32">
      <a href="https://travis-ci.com/github/alexellis/inlets-ghcr" target="_blank" rel="noopener nofollow ugc">travis-ci.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://travis-ci.com/github/alexellis/inlets-ghcr" target="_blank" rel="noopener nofollow ugc">Travis CI - Test and Deploy with Confidence</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by alexellis** [comment]
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/alexellis/inlets-ghcr" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/6358735/181100444-2e651a0e-b13a-4a16-918f-3e16e872f435.png" class="thumbnail onebox-avatar" width="382" height="382">

<h3><a href="https://github.com/alexellis/inlets-ghcr" target="_blank" rel="noopener nofollow ugc">alexellis/inlets-ghcr</a></h3>

<p>inlets-ghcr. Contribute to alexellis/inlets-ghcr development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by alexellis** [comment]
<p>Seems related - <a href="https://github.com/docker/for-linux/issues/396" rel="noopener nofollow ugc">https://github.com/docker/for-linux/issues/396</a></p>

**Comment by alexellis** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/fc1042df0bcc266af8731df14300077d990a9fae.png" data-download-href="/uploads/short-url/zXRa2UtrPh2Tq4iLzVYllcCFuEK.png?dl=1" title="Screenshot 2020-09-20 at 10.24.58" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/6358735/181100449-e2f83e1d-75c3-484a-aa7b-b3f8671d465b.png" alt="Screenshot 2020-09-20 at 10.24.58" data-base62-sha1="zXRa2UtrPh2Tq4iLzVYllcCFuEK" width="690" height="59" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/fc1042df0bcc266af8731df14300077d990a9fae_2_690x59.png, https://aws1.discourse-cdn.com/github/optimized/2X/f/fc1042df0bcc266af8731df14300077d990a9fae_2_1035x88.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/f/fc1042df0bcc266af8731df14300077d990a9fae_2_1380x118.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/fc1042df0bcc266af8731df14300077d990a9fae_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-09-20 at 10.24.58</span><span class="informations">1968×170 44.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>:man_facepalming: &lt;- this wasn’t required for the Docker Hub.</p>
